### PR TITLE
Manually skip workflow

### LINKED ISSUE #1354: Allow to manually skip (waiting) workflows
Add a button to skip a workflow that is currently waiting for e.g. `depends_on` or agent capacities.

### PR DESCRIPTION
**Pull request by the-writer-dev**
Trying to close #1354
 
@qwerty287 I had problems with the previous branch so moving to this one with resolution of your comments. Sorry for any potential confusion!

### CODE REVIEW

**Comment by bitethecode** [comment]
@qwerty287 Thanks for the comment! I will be working on them sooner

**Comment by bitethecode** [comment]
@qwerty287 Those should be resolved! 

Screen Shot 2022-12-14 at 1 21 24 PM

**Comment by lafriks** [comment]
would be nice if it would show confirmation dialog for workflow skip button

**Comment by bitethecode** [comment]
@lafriks Thanks for the comment! I will make sure to update that

**Comment by bitethecode** [comment]
Screen Shot 2022-12-15 at 12 22 48 PM

@lafriks does this look good?

**Comment by lafriks** [comment]
This also definetly helps, what I was more thinking about confirmation dialog with buttons yes/no or something like that to confirm before skipping steps

**Comment by bitethecode** [comment]
@lafriks Hmm.. I follow cancel flow. And it is not popping any dialog.. I wonder we want to throw a dialog for any mutating operations? From my understanding, skip workflow is way less impactful than cancel.

**Comment by lafriks** [comment]
That's definitely not requirement, just would be nice to have imho

Cancel button also would be nice to have that

**Comment by bitethecode** [comment]
@qwerty287 I agree..

**Comment by lafriks** [comment]
I have no problem with that 👍🏻

**Comment by gapodo** [comment]
> That's definitely not requirement, just would be nice to have imho
> 
> Cancel button also would be nice to have that

I'm with you on having those (I'd almost tend to see it as a requirement), just thinking about some long-running pipelines (stuff like integration testing, which takes potentially hours and "must" be run in the same workflow as the release build, or large builds that cannot be resumed part way), also re-running on accidentally canceled tags may cause troubles (i.e. things already uploaded to a, then canceling either the step or the entire pipeline and on a rerun running into conflics with leftovers from the last tagged run).

File: server/api/pipeline.go [hunk 1]
```diff
@@ -442,3 +442,26 @@ var deleteStr = `[
 		"out": "logs purged by %s on %s\n"
 	}
 ]`
+
+func PostSkipStep(c *gin.Context) {
```
Comment (6543) [inline server/api/pipeline.go:446]: step or workflow :D ...

please make a consistent naming acording to #745

Comment (bitethecode) [reply server/api/pipeline.go:446]: Okay.. which one do you think making more sense?

Comment (6543) [reply server/api/pipeline.go:446]: well the issue is about workflows that are pending

**Comment by 6543** [comment]
please resolve conflict :)

**Comment by anbraten** [comment]
closing as #1731 seems to be a newer version of this one

**Pull request closed without merging**